simplesam.py

```diff
@@ -76,10 +65,10 @@ def __init__(self, f, regions=False, kind=None, samtools_path=None):
             _, ext = os.path.splitext(f.name)
             if f.name == '<stdin>':  # stdin stream
                 self._sam_init(f)
-            elif ext == '.bam' or kind == 'bam':
+            elif ext == '.bam' or (kind is not None and kind.lower() == 'bam'):
```

Comment:
My gut feeling is that the extension should also be case-insensitive. Could you add that?